### PR TITLE
Making the code shorter and more readable

### PR DESCRIPTION
**Pull request by elichai**
Hi.
This is based and blocked by https://github.com/bitcoinops/bitcoin/pull/11
The point is to try and remove any unneeded complexity.
Solves #28 and fixed the assert in `verify_schnorr`

### CODE REVIEW

File: 0.2-Elliptic-Curve-Math.ipynb [hunk 1]
```diff
@@ -364,7 +344,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.6.8"
+   "version": "3.6.5"
```
Comment (elichai) [inline 0.2-Elliptic-Curve-Math.ipynb]: That's for the sake of consistency, all the rest of the notebooks share this version.

File: 0.2-Elliptic-Curve-Math.ipynb [hunk 2]
```diff
@@ -185,8 +175,8 @@
     "\n",
     "Consider:\n",
     "\n",
-    "* `a + b = b + a` over secp256k1\n",
-    "* `a * b = b * a` over secp256k1"
+    "* `a + b == b + a` over secp256k1\n",
```
Comment (elichai) [inline 0.2-Elliptic-Curve-Math.ipynb]: I'm now unsure about the change to double equality.
on one hand as a programmer it's easier to understand. on another hand when it comes as an equation it looks a bit weird.

File: 0.2-Elliptic-Curve-Math.ipynb [hunk 3]
```diff
@@ -108,15 +108,11 @@
...
     "\n",
     "# ECKey() operations\n",
-    "# convert our ints to 32 byte representations since that is what ECKey expects\n",
-    "a_bytes = a.to_bytes(32,'big')\n",
-    "b_bytes = b.to_bytes(32,'big')\n",
-    "\n",
     "# use our bytes to instantiate ECKey instances \n",
     "a_key = ECKey() \n",
-    "a_key.set(a_bytes, True) # set the secret value of the Eckey\n",
+    "a_key.set(a, True) # set the secret value of the Eckey\n",
```
Comment (bitschmidty) [inline 0.2-Elliptic-Curve-Math.ipynb]: Casing Eckey should be ECKey throughout

Comment (bitschmidty) [inline 0.2-Elliptic-Curve-Math.ipynb]: Getting `TypeError: object of type 'int' has no len()` running through this. Im on Taproot_V0.1.2.

File: 0.2-Elliptic-Curve-Math.ipynb (hunk 2 above)
Comment (bitschmidty) [reply 0.2-Elliptic-Curve-Math.ipynb]: I could go either way on this.

File: 0.2-Elliptic-Curve-Math.ipynb (hunk 1 above)
Comment (bitschmidty) [reply 0.2-Elliptic-Curve-Math.ipynb]: Not sure how this one got the wrong version, but thanks for finding it!

File: 0.2-Elliptic-Curve-Math.ipynb (hunk 3 above)
Comment (elichai) [reply 0.2-Elliptic-Curve-Math.ipynb]: Yeah it requires https://github.com/bitcoinops/bitcoin/pull/11

**Comment by jnewbery** [comment]
Rebased on master and added a few minor changes of my own. Thanks @elichai !

**Pull request merged**